### DISCUSSION TITLE
Is it possible for me to detect when someone joins or leaves a VC?

### DISCUSSION THREAD

**Discussion by ghost** [Rapptz/discord.py#7098] [Q&A]
I want to be able to tell if someone joins a VC and run a action after someone joins or leaves, sorry if this has been asked before I couldn't find it any where, how can I go through with this?

**Comment by lgaan** [comment]
This can be accomplished via [`on_voice_state_update`](https://discordpy.readthedocs.io/en/latest/api.html#discord.on_voice_state_update) . Check if `not after.voice` and `before.voice` for when they leave the channel; vice versa for joining.

**Comment by tnhnblgl** [comment]
`async def on_voice_state_update(self, member, before, after):
      if after.channel.id == channel_id and not member.bot:
         voice_client = await channel.connect()
         # Thing`
